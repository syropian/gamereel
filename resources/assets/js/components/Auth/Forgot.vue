<template>
  <div class="login-form w-3/4 mx-auto">
    <h3 class="text-brand text-2xl">Forgot your password?</h3>
    <p class="my-2 text-grey-dark">No problem! Enter your email below and we'll send you a link to reset your password.</p>
    <form class="my-6">
      <TextField name="email" class="mb-4" type="email" placeholder="Email Address" v-model="email"></TextField>
      <button type="submit" class="auth-button w-full h-10 bg-brand hover:bg-purple text-white rounded">Send Recovery Email</button>
    </form>
    <div class="rounded bg-grey-lighter p-2 text-sm text-black">All good? <router-link to="/auth/login" class="font-semibold text-brand hover:text-purple no-underline">Sign in!</router-link></div>
  </div>
</template>
<script>
import TextField from './TextField'
export default {
  name: 'Forgot',
  components: {
    TextField
  },
  data() {
    return {
      email: ''
    }
  }
}
</script>
<style lang="scss" scoped>
.auth-button {
  transition: background 150ms ease;
}
</style>

