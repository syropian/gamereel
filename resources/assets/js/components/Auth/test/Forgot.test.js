import { mount, shallow } from 'vue-test-utils'
import { createRenderer } from 'vue-server-renderer'
import Forgot from '../Forgot'

describe('Forgot Component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(Forgot)
  })

  it('sets the appropriate user data when the fields are typed in', () => {
    const vm = wrapper.vm
    type('jane@doe.com', 'input[name=email]')
    expect(vm.email).toBe('jane@doe.com')
  })

  it('matches the correct snapshot', () => {
    const renderer = createRenderer()
    const shallowWrapper = shallow(Forgot)
    renderer.renderToString(shallowWrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })

  const type = (text, selector) => {
    const node = wrapper.find(selector)
    node.element.value = text
    node.trigger('input')
  }
})
