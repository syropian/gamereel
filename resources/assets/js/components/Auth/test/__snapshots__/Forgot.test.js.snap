// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forgot Component matches the correct snapshot 1`] = `
<div class="login-form w-3/4 mx-auto">
    <h3 class="text-brand text-2xl">Forgot your password?</h3>
    <p class="my-2 text-grey-dark">No problem! Enter your email below and we'll send you a link to reset your password.</p>
    <form class="my-6">
        <!----><button type="submit" class="auth-button w-full h-10 bg-brand hover:bg-purple text-white rounded">Send Recovery Email</button></form>
    <div class="rounded bg-grey-lighter p-2 text-sm text-black">All good?
        <router-link to="/auth/login" class="font-semibold text-brand hover:text-purple no-underline">Sign in!</router-link>
    </div>
</div>
`;
